tests/subsys/llext/simple/src/test_llext_simple.c

```diff
@@ -437,6 +437,66 @@ ZTEST(llext, test_find_section)
 	zassert_equal(symbol_ptr, section_ptr,
 		      "symbol at %p != .data section at %p (%zd bytes in the ELF)",
 		      symbol_ptr, section_ptr, section_ofs);
+
+	llext_unload(&ext);
+}
+
+static LLEXT_CONST uint8_t test_detached_ext[] ELF_ALIGN = {
+	#include "detached_fn.inc"
+};
+
+static struct llext_loader *detached_loader;
+static struct llext *detached_llext;
+static elf_shdr_t detached_shdr;
+
+static bool test_section_detached(const elf_shdr_t *shdr)
+{
+	if (!detached_shdr.sh_addr) {
+		int res = llext_get_section_header(detached_loader, detached_llext,
+						   ".detach", &detached_shdr);
+
+		zassert_ok(res, "get_section_header should succeed");
+	}
+
+	return shdr->sh_name == detached_shdr.sh_name;
+}
+
+ZTEST(llext, test_detached)
+{
+	struct llext_buf_loader buf_loader =
+		LLEXT_BUF_LOADER(test_detached_ext, sizeof(test_detached_ext));
+	struct llext_load_param ldr_parm = LLEXT_LOAD_PARAM_DEFAULT;
+	int res;
+
+	ldr_parm.section_detached = test_section_detached;
+	detached_loader = &buf_loader.loader;
+
+	res = llext_load(detached_loader, "test_detached", &detached_llext, &ldr_parm);
+	zassert_ok(res, "load should succeed");
+
+#ifdef CONFIG_LLEXT_TYPE_ELF_SHAREDLIB
+	/*
+	 * Verify that the detached section is outside of the .text region.
+	 * This only works with the shared ELF type, because with other types
+	 * section addresses aren't "real," e.g. they can be 0.
```

Comment:
Since this test is only performed in the `WRITABLE` case, you could probably avoid this issue by comparing `sh_offset`s instead? Those should be accurate in every kind of ELF file.